snapcraft/shell_utils.py

```diff
@@ -38,4 +39,8 @@ def run_script(script, **kwargs):
         print("#!/bin/sh", file=tempf)
         print(script, file=tempf)
         tempf.flush()
-        return common.run_output(["/bin/sh", tempf.name], **kwargs)
+        if "SNAPCRAFT_DEBUG_SCRIPTS" in os.environ:
+                return common.run_output(["/bin/bash", "-x", tempf.name], **kwargs)
```

Comment:
any reason for switching from `/bin/sh` to `/bin/bash`?

I wonder if something like this would be appropriate under SNAPCRAFT_ENABLE_DEVELOPER_DEBUG rather than introducing a new variable?

Reply:
I need to use bash instead of sh, because sh doesn't support the `-x` option AFAIK (I'm wondering which one's better, `-x` (prints every command as executed) or `-v` (prints part of the script being executed as written in source)

`SNAPCRAFT_ENABLE_DEVELOPER_DEBUG` this exists? If so then that might be a better choice

Reply:
Note that that does print every command that's being executed in a script. This might be too verbose for some cases (since it makes the general debug log less useful), so a separate flag might make sense.
I'll leave that up to the team, but separating it makes sense IMO

Reply:
Note that that does print every command that's being executed in a script. This might be too verbose for some cases (since it makes the general debug log less useful), so a separate flag might make sense.
I'll leave that up to the team, but separating it makes sense IMO

Reply:
Makes sense, thanks.

`sh -x` should work. You may have unintended consequences by changing the shell, so I'd suggest trying it. Or simply writing the `set -x` at the top of the script.

Reply:
Also, your formatting needs to be corrected (see static test output).

Reply:
I've turned it into setting -x inline instead of changing the shell
The variable name's fine then?

Reply:
Sorry I dropped the ball on this thread.

@sergiusens Would you agree that it would be fair to enable this for SNAPCRAFT_ENABLE_DEVELOPER_DEBUG?